XenAdmin/Diagnostics/Checks/ServerSelectionCheck.cs

```diff
@@ -61,7 +61,8 @@ protected override Problem RunCheck()
             if (update == null || !update.EnforceHomogeneity()) 
                 return null;
 
-            if (pool.Connection.Cache.Hosts.Any(h => !update.AppliedOn(h) && !selectedServers.Contains(h)))
+            //The check is hit when there is a patch-non-applied host, whose version is equal to the selected host but not selected
+            if (pool.Connection.Cache.Hosts.Any(h => !update.AppliedOn(h) && h.ProductVersion() == selectedServers[0].ProductVersion() && !selectedServers.Contains(h)))

```

Comment:
Shouldn't we be using the method Helpers.productVersionCompare() for the comparison instead of comparing the strings?